src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let localSep = separator;
+    let result = '';
+
+    if (localSep === null) {
+      localSep = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i !== this.length - 1) {
+        result += this[i] + localSep.toString();
```

Comment:
```suggestion
        result += '' + this[i] + localSep;
```
подумай чому це працюватиме без if зверху